### DISCUSSION TITLE
@types/gulp-muxml

### DISCUSSION THREAD

**Discussion by Dulux-Oz** [DefinitelyTyped/DefinitelyTyped#58111] [Request a new @types package] [answered]
Hi All,

Could someone help a (TypeScript) noob with some advice, please, by answering one (or more) of the following questions:

1. Does anyone have the TypeScript def file for gulp-muxml, and may I please have a copy?
2. Could someone please write the TypeScript def file for gulp-muxml?
3. Could someone help me to write the TypeScript def file for gulp-muxml?
4. Could someone point me towards a different XML minifier that works with Gulp and which is TypeScript-useable?
5. Could someone point me towards how to use gulp-muxml "as it is" with TypeScript?

Thanks in advance,

Dulux-Oz

**Comment by martin-badin** [comment, marked as answer]
PR is ready #58114

**Comment by Dulux-Oz** [comment]
Thanks Martin, for both creating the .d.ts file and for doing it so quickly - I really appreciate both.

Can I ask, in an effort to educate myself, how you went about doing this (ie the process you went through) - it might be better to answer this over email (matthew@peregrineit.net) - thanks.

Dulux-Oz

**Reply by martin-badin** [reply]
First, you need to read this readme how to create a new package
https://github.com/DefinitelyTyped/DefinitelyTyped#create-a-new-package

Then, when I want to create a definition for a new package so I first follow the documentation, and then I read the source code (It is better because not all of the methods/options are documented).

When I want to create a definition I follow this pattern
```
declare namespace muxml {}

declare function muxml(): any;
// or
declare const muxml: any;
// or
declare class Muxml {}

export = muxml;
```

**Reply by Dulux-Oz** [reply]
Thanks :-)
